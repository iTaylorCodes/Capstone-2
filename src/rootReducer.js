import { v4 as uuidv4 } from "uuid";

const INITIAL_STATE = {
  accountWasDeleted: false,
  recentSearches: [],
};

const rootReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case "ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: true,
      };

    case "RESET_ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: false,
      };

    case "NEW_SEARCH":
      if (state.recentSearches.length >= 5) {
        state.recentSearches.shift();
      }

      return {
        ...state,
        recentSearches: [
          ...state.recentSearches,
          {
            city: action.searchTerm,
            image: action.image,
            scores: action.scores,
            id: uuidv4(),
          },
        ],
      };

    default:
      return state;
  }
};

export default rootReducer;
