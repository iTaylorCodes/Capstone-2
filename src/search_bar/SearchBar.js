import { useState } from "react";
import { useDispatch } from "react-redux";
import { PlacesApi } from "../api/api";

const SearchBar = ({ setSearchError }) => {
  const [searchTerm, setSearchTerm] = useState("");
  const dispatch = useDispatch();

  const searchFor = async (searchTerm) => {
    let image;
    try {
      image = await PlacesApi.getImage(searchTerm);
    } catch (e) {
      console.error(e);
      setSearchError(e.message);
      return;
    }

    dispatch({
      type: "NEW_SEARCH",
      searchTerm,
      image,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!searchTerm) return;

    searchFor(searchTerm.trim());
    setSearchTerm("");
  };

  const handleChange = (e) => {
    setSearchTerm(e.target.value);
    setSearchError(null);
  };

  return (
    <div className="mb-3">
      <form className="input-group" onSubmit={handleSubmit}>
        <div className="form-floating">
          <input
            className="form-control ps-5 pe-5"
            name="searchTerm"
            id="city-search"
            placeholder="Search City, State:"
            value={searchTerm}
            onChange={handleChange}
          />
          <label htmlFor="city-search">Search City, State:</label>
        </div>
        <button
          className="btn btn-primary"
          type="submit"
          data-testid="search-button"
        >
          <i className="fa fa-search"></i>
        </button>
      </form>
    </div>
  );
};

export default SearchBar;
