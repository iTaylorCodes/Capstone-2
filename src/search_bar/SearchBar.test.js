import SearchBar from "../search_bar/SearchBar";
import { render, act, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../rootReducer";

const store = createStore(rootReducer);

const setSearchError = jest.fn();

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <SearchBar setSearchError={setSearchError} />
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <SearchBar setSearchError={setSearchError} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("resets searchbar value", async () => {
  await act(async () => {
    const { getByTestId, getByLabelText } = render(
      <Provider store={store}>
        <SearchBar setSearchError={setSearchError} />
      </Provider>
    );

    const searchInput = getByLabelText("Search City, State:");
    await fireEvent.change(searchInput, {
      target: { value: "Los Angeles, CA" },
    });

    expect(getByLabelText("Search City, State:").value).toBe("Los Angeles, CA");

    const button = getByTestId("search-button");
    await fireEvent.click(button);

    expect(getByLabelText("Search City, State:").value).toBe("");
  });
});
