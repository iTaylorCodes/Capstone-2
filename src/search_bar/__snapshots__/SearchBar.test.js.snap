// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <form
      class="input-group"
    >
      <div
        class="form-floating"
      >
        <input
          class="form-control ps-5 pe-5"
          id="city-search"
          name="searchTerm"
          placeholder="Search City, State:"
          value=""
        />
        <label
          for="city-search"
        >
          Search City, State:
        </label>
      </div>
      <button
        class="btn btn-primary"
        data-testid="search-button"
        type="submit"
      >
        <i
          class="fa fa-search"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
